### PR TITLE
Remove Werkzeug limitation after flask-login was fixed

### PR DESCRIPTION
**Pull request by potiuk**
The Werkzeug limitation removed in #25270 can be removed now, when
Flask-login is fixed and 0.6.2 version is released that supports it.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by josh-fell — APPROVED** [review]

**Review by jedcunningham — APPROVED** [review]

**Comment by potiuk** [comment]
Let's not merge this one (yet) flask login 0.6.2 break one of our tests (Google Oauth login). I can reliably reprosuce it but i need to find out if it is a real bug or test issue (i presume the latter - seems like we are mocking some internal call that have changed)

**Comment by potiuk** [comment]
Actually, this turned to be real error @josh-fell and @jedcunningham - https://github.com/apache/airflow/pull/25318

I think this one actually can have quite a bad effect on our users, I will follow up with flask-login maintainer to verify the actual scope of the problem.

As I see now, some of our users will likely have google openid auth backend stop working if they switch to Flask-Login 0.6.2 (And there is nothing to prevent them from doing so except constraints). 

However in the meantime we should merge #25318 to stop main from failing

**Comment by potiuk** [comment]
Right. I also had to fix the tests because Werkzeug 2.2 suddenly started to return &#39; instead of &#x27; :shrug:

**Pull request merged**
